File: Dal/Models/ExamPeriod.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DAL.Models
{ 
    public class ExamPeriod
    {
        public int ExamPeriodId { get; set; }
        public string Name { get; set; }

        public int YearId { get; set; }
        public Year Year { get; set; }

        public ICollection<DAL.Models.PresentationDay>? PresentationDays { get; set; } = new List<DAL.Models.PresentationDay>();
    }
}

File: Client/src/app/shared/models/examperiod.ts
// Note: To avoid circular reference issues, import Year and PresentationDay only where needed.

export interface ExamPeriod {
  examPeriodId: number;
  name: string;
  yearId: number;
  year: any; // Use 'any' to avoid circular reference for now
  presentationDays?: any[]; // Use 'any[]' to avoid circular reference for now
}

export interface ExamPeriodCreateDto {
  name: string;
  yearId: number;
}

export interface ExamPeriodUpdateDto {
  name: string;
  yearId: number;
}
